projects/plugins/wpcomsh/feature-plugins/hooks.php

```diff
@@ -347,3 +347,63 @@ function wpcomsh_remove_jetpack_manage_menu_item() {
 }
 
 add_action( 'admin_menu', 'wpcomsh_remove_jetpack_manage_menu_item', 1001 ); // Automattic\Jetpack\Admin_UI\Admin_Menu uses 1000.
+
+/**
+ * Prevent plugin activation if it would exceed the plan's plugin limit.
+ *
+ * @param array $new_value The new value of the option.
+ * @param array $old_value The old value of the option.
+ * @return array|WP_Error The filtered value.
+ */
+function wpcomsh_prevent_exceeding_plugin_limit( $new_value, $old_value ) {
+	if ( is_multisite() && is_network_admin() ) {
+		return $new_value;
+	}
+
+	$newly_activated = array_diff( $new_value, $old_value );
+	if ( empty( $newly_activated ) ) {
+		return $new_value;
+	}
+
+	// Plugins are manually activated one by one
```

Comment:
I'll need to see how plugins are activated during provisioning to make sure this code doesn't interfere.